libraries/AP_InertialSensor/AP_InertialSensor_RST.cpp

```diff
@@ -0,0 +1,401 @@
+#define GYRO_INT1_TSH_YH          0x34
+#define GYRO_INT1_TSH_YL          0x35
+#define GYRO_INT1_TSH_ZH          0x36
+#define GYRO_INT1_TSH_ZL          0x37
+#define GYRO_INT1_DURATION        0x38
+#define GYRO_LOW_ODR              0x39
+
+
+/* Internal configuration values */
+#define GYRO_REG1_POWER_NORMAL    (1<<3)
+#define GYRO_REG1_Z_ENABLE        (1<<2)
+#define GYRO_REG1_Y_ENABLE        (1<<1)
+#define GYRO_REG1_X_ENABLE        (1<<0)
+
+#define GYRO_REG4_BLE             (1<<6)
+
+#define GYRO_REG5_FIFO_ENABLE     (1<<6)
+#define GYRO_REG5_REBOOT_MEMORY   (1<<7)
+
+#define GYRO_STATUS_ZYXOR         (1<<7)
+#define GYRO_STATUS_ZOR           (1<<6)
+#define GYRO_STATUS_YOR           (1<<5)
+#define GYRO_STATUS_XOR           (1<<4)
+#define GYRO_STATUS_ZYXDA         (1<<3)
+#define GYRO_STATUS_ZDA           (1<<2)
+#define GYRO_STATUS_YDA           (1<<1)
+#define GYRO_STATUS_XDA           (1<<0)
+
+#define GYRO_FIFO_CTRL_BYPASS_MODE              (0<<5)
+#define GYRO_FIFO_CTRL_FIFO_MODE                (1<<5)
+#define GYRO_FIFO_CTRL_STREAM_MODE              (1<<6)
+#define GYRO_FIFO_CTRL_STREAM_TO_FIFO_MODE      (3<<5)
+#define GYRO_FIFO_CTRL_BYPASS_TO_STREAM_MODE    (1<<7)
+
+#define GYRO_DEFAULT_RATE               800 //data output frequency
+#define GYRO_DEFAULT_RANGE_DPS          2000
+#define GYRO_DEFAULT_FILTER_FREQ        35
+#define GYRO_TEMP_OFFSET_CELSIUS        40
+
+
+// constructor
+AP_InertialSensor_RST::AP_InertialSensor_RST(AP_InertialSensor &imu,
+                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_gyro,
+                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_accel,
+                              enum Rotation rotation_g,
+                              enum Rotation rotation_a)
+    : AP_InertialSensor_Backend(imu)
+    , _dev_gyro(std::move(dev_gyro))
+    , _dev_accel(std::move(dev_accel))
+    , _rotation_g(rotation_g)
+    , _rotation_a(rotation_a)
+{
+}
+
+AP_InertialSensor_RST::~AP_InertialSensor_RST()
+{
+}
+
+/*
+  detect the sensor
+ */
+AP_InertialSensor_Backend *AP_InertialSensor_RST::probe(AP_InertialSensor &imu,
+                                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_gyro,
+                                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_accel,
+                                              enum Rotation rotation_g,
+                                              enum Rotation rotation_a)
+{
+    if (!dev_gyro && !dev_accel) {
+        return nullptr;
+    }
+    AP_InertialSensor_RST *sensor
+        = new AP_InertialSensor_RST(imu, std::move(dev_gyro), std::move(dev_accel), rotation_g, rotation_a);
+    if (!sensor || !sensor->_init_sensor()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+/*
+ * init gyro
+ */
+bool AP_InertialSensor_RST::_init_gyro(void)
+{
+
+    _gyro_spi_sem = _dev_gyro->get_semaphore();
+
+    if (!_gyro_spi_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
```

Comment:
`_gyro_spi_sem` isn't used anywhere else except in this function (and shouldn't actually be used). Please remove it from the class and do like this here:

```c++
if (!_dev_gyro->get_semaphore()->take(...)) {
```